src/mainwindow.cpp

```diff
@@ -131,13 +131,10 @@ MainWindow::MainWindow(QWidget *parent)
     // static reference to us
     s_self = this;
 
-    // handle logging as signal/slot to even more prevent crashes when
```

Comment:
And there will be no crashes because something gets logged while shutting down?

Reply:
One case to try this (haven't done so myself yet,) would be to log in the mainWindow destructor. If it survives that, we are safe.

Reply:
As long as the widget is alive, it shouldn't crash. The Qpointer usage is here for exactly this reason. On destruction it should become null and no further log calls will be processed.

Reply:
Tried it, works okay. Here's a log ( I added an else case where we call `log()`).

```
Warning: deleting mainwindow (/home/waqar/projects/QOwnNotes/src/mainwindow.cpp:726, virtual MainWindow::~MainWindow())
Got log request
Warning: Deleting log widget (/home/waqar/projects/QOwnNotes/src/widgets/logwidget.cpp:99, virtual LogWidget::~LogWidget())
Got log request
Debug: deleting 0x560d8cb97b68 for "en_US"
log widget is dead now
Debug: deleting 0x560d8c726f20 for "en_US"
log widget is dead now
Debug: Removing loader:  Sonnet::Loader(0x560d8c0de720)
log widget is dead now
```

"Log widget is dead now" this is the else case, where the log widget has long gone.